src/audio/audio_manager.cpp

```diff
@@ -16,17 +21,74 @@ AudioManager::AudioManager() : muted(true) {
         if (!music.openFromFile(filename)) {
             std::cerr << "AudioManager: Cannot open " << filename << std::endl;;
         }
-        music.setLoop(true);
-        if (!muted)
+
+        music.setVolume(d.volume);
+        music.setLoop(d.loop);
+
+        if (!muted) {
             music.play();
+        }
+    });
+
+    events::sound_effects_event.connect([&](const events::AudioData &d) {
+        std::string filename = d.filename;
+
+        sounds[filename]->setVolume(d.volume);
+        sounds[filename]->setLoop(d.loop);
+
+        if (!muted) {
+            sounds[filename]->play();
+        }
+        else if (muted && d.loop) {
+            // Change state to paused so that when audio is un-muted, the looped sound will play
+            sounds[filename]->play();
+            sounds[filename]->pause();
+        }
+    });
+
+    events::stop_sound_effects_event.connect([&](const std::string filename) {
+        sounds[filename]->stop();
     });
 
     events::toggle_mute_event.connect([&]() {
         muted = !muted;
         if (muted) {
             music.pause();
-        } else {
+
+            // Mute all sound effects
+            for (auto it = sounds.begin(); it != sounds.end(); it++) {
+                it->second->pause();
+            }
+        }
+        else {
             music.play();
+
+            // Play looped sound effects that has been paused
+            for (auto it = sounds.begin(); it != sounds.end(); it++) {
+                if (it->second->getStatus() == sf::SoundSource::Status::Paused) {
+                    it->second->play();
+                }
+            }
         }
     });
 }
+
+void AudioManager::loadSound(sf::SoundBuffer * soundBuffer, std::string filename) {
+    if (!soundBuffer->loadFromFile(filename)) {
+        std::cerr << "AudioManager: Cannot open" << filename << std::endl;;
+    }
+}
+
+void AudioManager::loadSounds() {
+    // Load sound effects into memory
+
+    // BUILD_CONFIRM_SOUND
```

Comment:
This will get bloated. Can we automate this with a loop through the sounds directory, like [we do for the asset loader](https://github.com/9hack/aldente/blob/aa602252899b6e1b61ea82499e6c358a0b474c28/src/asset_loader.cpp#L28)? We can then have maps of filename -> buffer, and filename -> sound, instead of bloating up the manager with members per sound.